src/perturbopy/postproc/calc_modes/spectral_cumulant.py

```diff
@@ -0,0 +1,88 @@
+import numpy as np
+import h5py as h5
+import matplotlib.pyplot as plt
+from perturbopy.postproc.calc_modes.calc_mode import CalcMode
+
+
+class SpectralCumulant(CalcMode):
+    """
+    A class to handle and analyze spectral functions computed using the cumulant expansion approach.
+
+    Attributes:
+    - T (numpy.ndarray): Array of temperatures corresponding to the spectral data.
+    - w (numpy.ndarray): Array of energy values (ω) in electron volts (eV).
+    - dw (float): Energy step size for the ω grid in eV.
+    - Akw (numpy.ndarray): Spectral function data, indexed by k-point, band, temperature, and ω.
+    """
+    def __init__(self, prefix):
+        f = h5.File(f"{prefix}_spectral_cumulant.h5", "r")
+        Akw = f['spectral_functions']
+        w_lower = np.asarray(f['w_lower_index'])
+        w_upper = np.asarray(f['w_upper_index'])
+        dw = np.asarray(f['wfreq_step_eV'])
+        self.T = np.asanyarray(f['temperatures'])
+        self.w = np.arange(w_lower, w_upper + 1) * dw
+        self.dw = dw
+        Akw_np = []
+        for key in Akw.keys():
+            Akw_np.append(np.asarray(Akw[key]))
+        f.close()
+        self.Akw = np.asarray(Akw_np)
+
+    def plot_Aw(self, ik=0, it=0, ib=0):
+        if (it > len(self.T)):
+            print('Temperature index is out of range')
```

Comment:
use `raise RuntimeError('message')
instead of print(); return

or warnings.warn('')